core/src/main/scala/com/mesosphere/usi/core/protos/ProtoBuilders.scala

```diff
@@ -273,4 +274,25 @@ private[usi] object ProtoBuilders {
       .setOffers(MesosEvent.Offers.newBuilder().addOffers(offer))
       .build()
   }
+
+  def newDockerImage(dockerImageName: String): Image = {
+    Image
+      .newBuilder()
+      .setType(Image.Type.DOCKER)
+      .setDocker(Image.Docker.newBuilder().setName(dockerImageName))
+      .build()
+  }
+
+  def newContainerInfo(imageName: Option[String]): Option[Mesos.ContainerInfo] = {
+    imageName match {
+      case Some(imageName) =>
+        Some(
+          ContainerInfo
+            .newBuilder()
+            .setType(Mesos.ContainerInfo.Type.MESOS)
+            .setMesos(ContainerInfo.MesosInfo.newBuilder().setImage(newDockerImage(imageName)).build())
+            .build())
+      case _ => None
```

Comment:
This would be a map operation
```scala
imageName.map { name =>
          ContainerInfo
            .newBuilder()
            .setType(Mesos.ContainerInfo.Type.MESOS)
            .setMesos(ContainerInfo.MesosInfo.newBuilder().setImage(newDockerImage(name)).build())
            .build())
}
```